File: easy/leetcode2.py
class Solution:
    def reverse(self, x: int) -> int:
        b = 0
        if x < 0:
            b, x = 1, abs(x)
        mod, res = 0, 0
        while x:
            #下面两行分离出每一个位，从而完成逆序
            x, mod = x // 10, x % 10
            res = res * 10 + mod
            if res > 2147483648:
                return 0
        if b == 1:
            res = -res
        return res

File: medium/leetcode2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int anathorAdd = 0;
        int res = 0;
        ListNode node = new ListNode(0);
        ListNode tmp = node;
        while (l1 != null || l2 != null) {
            int num = 0;
            if (l1 != null) {
                num+=l1.val;
                l1 = l1.next;
            }
            if (l2 != null) {
                num+=l2.val;
                l2 = l2.next;
            }
            num+=anathorAdd;
            anathorAdd = num/10;
            num %= 10;
            tmp.next = new ListNode(num);
            tmp = tmp.next;
        }
        if (anathorAdd == 1) {
            tmp.next = new ListNode(1);
        }
        return node.next;
    }
}